packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -302,9 +304,18 @@ async function createTreeCodeFromPath(
 
       const parallelSegmentPath = subSegmentPath.join('/')
 
+      const fileTypesToResolve: (FileTypeValues | 'page')[] =
+        Object.values(FILE_TYPES)
+
+      // Special case for exports in parent pages
```

Comment:
This just fixes a special case (only for dynamic routes). Might be better to do a more extensive code rewrite on how pages in parent routes are added to the tree code.